### DISCUSSION TITLE
LSposed on Vivo X80 Pro (Android 12)

### DISCUSSION THREAD

**Discussion by washo4evr** [LSPosed/LSPosed#2089] [General]
Hi everyone,
after unlocking bootloader and installing Magisk 25.2 (Zygisk activated), I tried installing LSposed 1.8.3
I can see it under the Magisk modules list and the icon was properly created on the main page but when I tap the icon, nothing happens.

here are the logs from LSposed
[log.zip](https://github.com/LSPosed/LSPosed/files/9408263/log.zip)

Happy to upload any other files if needed too
Thanks for your help

edit: installed app setting reborn 1.6 and got a notification that the module wasn't activated but nothing happened when I tapped the notification

**Comment by washo4evr** [comment]
just tried the 1.8.3 (6599) debug and same happen

**Reply by washo4evr** [reply]
I spent some time looking at the logs but couldn't figure out what is going on.

Could someone help me on this?

Thanks

**Reply by mywalkb** [reply]
you should install latest debug version and post log, from your previous logs there is nothing useful for debug the issue.
You can follow [this](https://github.com/topjohnwu/Magisk/issues/6042#issuecomment-1167132605) comment for catch logs.

**Reply by washo4evr** [reply]
[boot-2022-08-31-10-32-56.zip](https://github.com/LSPosed/LSPosed/files/9463732/boot-2022-08-31-10-32-56.zip)
Awesome
I flashed this module and attached the output here

I noticed a lot of errors related to SElinux but not sure how to fix

**Reply by mywalkb** [reply]
> I noticed a lot of errors related to SElinux but not sure how to fix

something is wrong with your magisk installation, not only lsposed 

```
[    5.690269][    T1] selinux: SELinux: Could not set context for /data/adb/modules:  Permission denied
[    5.691500][    T1] selinux: SELinux: Could not set context for /data/adb/modules/hosts:  Permission denied
[    5.691953][    T1] selinux: SELinux: Could not set context for /data/adb/modules/hosts/system:  Permission denied
[    5.692393][    T1] selinux: SELinux: Could not set context for /data/adb/modules/hosts/system/etc:  Permission denied
[    5.692606][    T1] selinux: SELinux: Could not set context for /data/adb/modules/hosts/system/etc/hosts:  Permission denied
[    5.692693][    T1] selinux: SELinux: Could not set context for /data/adb/modules/hosts/module.prop:  Permission denied
[    5.693002][    T1] selinux: SELinux: Could not set context for /data/adb/modules/safetynet-fix:  Permission denied
[    5.693395][    T1] selinux: SELinux: Could not set context for /data/adb/modules/safetynet-fix/system.prop:  Permission denied
[    5.693483][    T1] selinux: SELinux: Could not set context for /data/adb/modules/safetynet-fix/service.sh:  Permission denied
```
All other magisk modules works?

**Reply by mywalkb** [reply]
maybe is related to this issue https://github.com/topjohnwu/Magisk/issues/5148

**Reply by washo4evr** [reply]
good point, I couldn't get jamesDSP or any other modules to work though

**Reply by mywalkb** [reply]
I did not understand, only lsposed does not work or the others as well as safetynet-fix which uses zygisk?
If is only lsposed, could you post also log.zip of debug version? Otherwise is better open an issue on magisk.

**Reply by washo4evr** [reply]
I can pass the safety check
several modules installed are working, others are not
the most visible non working is LSposed where the icon appears but nothing else happens
I have similar experience with viperFX/jamesDSP but I thought it was more a driver related issue. Could very well be a Magisk issue
I created an issue on XDA about that https://forum.xda-developers.com/t/support-for-vivo-x80-pro.4487185/

**Reply by mywalkb** [reply]
I repeat the previous request:

> If is only lsposed, could you post also log.zip of debug version?

For issue I think is better open [here](https://github.com/topjohnwu/Magisk/issues/new/choose)

**Reply by washo4evr** [reply]
[log.zip](https://github.com/LSPosed/LSPosed/files/9472094/log.zip)
this is the zip from /data/adb/lspd/log/ when using debug version 1.8.3.6599

**Reply by mywalkb** [reply]
I hope @yujincheng08 can analyze logs better than me.

```
[ 2022-09-01T08:45:29.117        0:  1205:  1205 W/LSPosedDex2Oat  ] Check mount failed for /apex/com.android.art/bin/dex2oat32
[ 2022-09-01T08:45:29.121        0:  1957:  1205 I/LSPosed         ] Enable dex2oat wrapper
[ 2022-09-01T08:45:29.122        0:  1205:  1205 D/LSPosedDex2Oat  ] Check mount succeeded
[ 2022-09-01T08:45:29.123        0:  1205:  1898 D/LSPosedService  ] LSPSystemServerService::LSPSystemServerService onRegistration: serial org.lsposed.lspd.service.LSPSystemServerService@4df307b
[ 2022-09-01T08:45:38.727     1000:  2923:  4110 D/LSPosed Bridge  ] onTransact: action=ACTION_GET_BINDER, callingUid=10092, callingPid=6530
[ 2022-09-01T08:45:38.727        0:  1205:  1920 D/LSPosedService  ] Skipped com.vivo.cota/10092
[ 2022-09-01T08:45:38.727    10092:  6530:  6530 D/LSPosed         ] magisk_loader.cpp:231#void lspd::MagiskLoader::OnNativeForkAndSpecializePost(JNIEnv *, jstring): skipped com.vivo.cota
[ 2022-09-01T08:45:40.858        0:  2930:  2931 D/LSPosed         ] zygisk_main.cpp:362#void lspd::CompanionEntry(int): Got cache with fd=5 size=24
[ 2022-09-01T08:46:05.066        0:  2930:  2931 D/LSPosed         ] zygisk_main.cpp:362#void lspd::CompanionEntry(int): Got cache with fd=5 size=24
[ 2022-09-01T08:46:05.066        0: 12141: 12141 D/LSPosed         ] symbol_cache.cpp:55#void lspd::InitSymbolCache(lspd::SymbolCache *): InitSymbolCache
[ 2022-09-01T08:46:05.066        0: 12141: 12141 D/LSPosed         ] symbol_cache.cpp:57#void lspd::InitSymbolCache(lspd::SymbolCache *): Already initialized
[ 2022-09-01T08:46:06.082        0:  2930:  2931 D/LSPosed         ] zygisk_main.cpp:362#void lspd::CompanionEntry(int): Got cache with fd=5 size=24
[ 2022-09-01T08:46:06.082        0: 12271: 12271 D/LSPosed         ] symbol_cache.cpp:55#void lspd::InitSymbolCache(lspd::SymbolCache *): InitSymbolCache
[ 2022-09-01T08:46:06.082        0: 12271: 12271 D/LSPosed         ] symbol_cache.cpp:57#void lspd::InitSymbolCache(lspd::SymbolCache *): Already initialized
[ 2022-09-01T08:46:11.160     1001: 12604: 12604 D/LSPosed         ] magisk_loader.cpp:231#void lspd::MagiskLoader::OnNativeForkAndSpecializePost(JNIEnv *, jstring): skipped com.qualcomm.qti.modemtestmode
[ 2022-09-01T08:46:29.114        0:  2930:  2931 D/LSPosed         ] zygisk_main.cpp:362#void lspd::CompanionEntry(int): Got cache with fd=5 size=24
[ 2022-09-01T08:46:29.114        0: 14624: 14624 D/LSPosed         ] symbol_cache.cpp:55#void lspd::InitSymbolCache(lspd::SymbolCache *): InitSymbolCache
[ 2022-09-01T08:46:29.115        0: 14624: 14624 D/LSPosed         ] symbol_cache.cpp:57#void lspd::InitSymbolCache(lspd::SymbolCache *): Already initialized
```

I would try to temporarily disable selinux with `setenforce 0`, just for test.

**Reply by washo4evr** [reply]
thanks so much for looking into this

so, I was able to setenforce 0, and then getenforce properly reports 'Permissive'
this gets reset  when I reboot, so I tried a magisk module to force it to stay in permissive mode
but, looking at the logs, it seems like this module is executed after the rest, so, despite having a permissive SELinux, I'm still getting the same error

is there a way to permanently change this and/or change the order at which magisk modules run?

**Reply by mywalkb** [reply]
> is there a way to permanently change this and/or change the order at which magisk modules run?

with android is not easy todo and explain.
It's just a test, you can change the start script of lsposed.

```
cp /data/adb/modules/zygisk_lsposed/service.sh  /data/adb/modules/zygisk_lsposed/service.sh.orig

cat > /data/adb/modules/zygisk_lsposed/service.sh << EOF
/system/bin/setenforce 0

MODDIR=${0%/*}
cd "$MODDIR"
# post-fs-data.sh may be blocked by other modules. retry to start this
unshare -m sh -c "$MODDIR/daemon --from-service $@&"
EOF
```

after the test is ended you can restore with

```
cp /data/adb/modules/zygisk_lsposed/service.sh.orig /data/adb/modules/zygisk_lsposed/service.sh
```

**Reply by washo4evr** [reply]
Hi,
[log.zip](https://github.com/LSPosed/LSPosed/files/9479944/log.zip)
[boot-2022-09-02-10-10-20.zip](https://github.com/LSPosed/LSPosed/files/9479945/boot-2022-09-02-10-10-20.zip)

I tried this and the same happens
here are the logs from both logcatcher and LSposed (looks similar to before)

seems like the issue is more related to Magisk and SElinux. what do you think?

**Reply by mywalkb** [reply]
> seems like the issue is more related to Magisk and SElinux. what do you think?

SELinux should be disabled. If the issue is related to Magisk also safetynet-fix wasn't supposed to work. safetynet-fix use zygisk as LSPosed.

Could you post the output of this command?

`ls -lZR /data/adb`

if something is wrong we can try to restore with `magisk --restorecon`

**Reply by washo4evr** [reply]
hey
[command.txt](https://github.com/LSPosed/LSPosed/files/9480890/command.txt)

this command ran
here is the output

**Reply by mywalkb** [reply]
also context on files are correct, I can't find wrong context.
Could you try to disable all modules except LSPosed?

**Reply by washo4evr** [reply]
so, to be sure, I completely uninstalled all modules and Magisk
reinstalled magisk 25.2, Zygisk + denyList
installed logCatcher (attached as 'before_LSposed.log') + LSposed 1.8.3.6600 (with the selinux modification)
LSposed icon is properly created but will not run

**Reply by washo4evr** [reply]
[log.zip](https://github.com/LSPosed/LSPosed/files/9483065/log.zip)
[after_LSposed.log.log](https://github.com/LSPosed/LSPosed/files/9483066/after_LSposed.log.log)
[before_LSposed.log.log](https://github.com/LSPosed/LSPosed/files/9483067/before_LSposed.log.log)

**Reply by washo4evr** [reply]
for the sake of trying, I went ahead, uninstalled/cleaned and tried the most recent Magisk canary (25.203)
same results
[after_LSposed_canary.log](https://github.com/LSPosed/LSPosed/files/9483106/after_LSposed_canary.log)
[log_canary.zip](https://github.com/LSPosed/LSPosed/files/9483108/log_canary.zip)

really not sure what to try next

**Reply by mywalkb** [reply]
> really not sure what to try next

have you tried old versions? 1.7.2 or 1.8.0 before dex2oat is introduced.

**Comment by washo4evr** [comment]
this is so bizarre as I can run 'su', edit build.props, access root (using total commander), and adaway is working too

**Comment by yujincheng08** [comment]
You should increase enable log in dev setting and increase the log buffer. Otherwise, no useful log can be obtained.

**Comment by washo4evr** [comment]
Thanks for the suggestion.
do you mean the "Logger buffer sizes"?
I had 256K and I changed it to 8M

which setting is the 'enable log'? I have one setting at the bottom of the page under Autofill called logging level. is that it?
